internal/alias/target/alias_test.go

```diff
@@ -0,0 +1,472 @@
+	t.Run("with destination", func(t *testing.T) {
+		a, err := NewAlias(context.Background(), "global", "with.destination", WithDestinationId("ttcp_1234567890"))
+		require.NoError(t, err)
+		assert.NotNil(t, a)
+		assert.Equal(t, a.ScopeId, "global")
+		assert.Equal(t, a.Value, "with.destination")
+		assert.Equal(t, a.DestinationId, "ttcp_1234567890")
+	})
+}
+
+func TestCreate(t *testing.T) {
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	ctx := context.Background()
+	wrapper := db.TestWrapper(t)
+
+	iamRepo := iam.TestRepo(t, conn, wrapper)
+	_, proj := iam.TestScopes(t, iamRepo)
+	tar := tcp.TestTarget(ctx, t, conn, proj.GetPublicId(), "test")
+
+	cases := []struct {
+		name        string
+		scope       string
+		value       string
+		opts        []Option
+		validate    func(*testing.T, *Alias)
+		errContains string
+	}{
+		{
+			name:  "valid",
+			scope: "global",
+			value: "valid.alias",
+			opts:  []Option{WithDestinationId(tar.GetPublicId())},
+			validate: func(t *testing.T, a *Alias) {
+				t.Helper()
+				assert.Equal(t, a.DestinationId, tar.GetPublicId())
+			},
+		},
+		{
+			name:  "valid with host",
+			scope: "global",
+			value: "host.valid.alias",
+			opts:  []Option{WithDestinationId(tar.GetPublicId()), WithHostId("hst_1234567890")},
+			validate: func(t *testing.T, a *Alias) {
+				t.Helper()
+				assert.Equal(t, a.DestinationId, tar.GetPublicId())
+				assert.Equal(t, a.HostId, "hst_1234567890")
+			},
+		},
+		{
+			name:  "valid no destination",
+			scope: "global",
+			value: "nodestination.alias",
+			validate: func(t *testing.T, a *Alias) {
+				t.Helper()
+				assert.Empty(t, a.DestinationId)
+			},
+		},
+		{
+			name:  "valid with name",
+			scope: "global",
+			value: "valid-with-name.alias",
+			opts:  []Option{WithName("valid-with-name")},
+			validate: func(t *testing.T, a *Alias) {
+				t.Helper()
+				assert.Equal(t, "valid-with-name", a.Name)
+			},
+		},
+		{
+			name:  "valid with description",
+			scope: "global",
+			value: "valid-with-description.alias",
+			opts:  []Option{WithName("valid-with-description"), WithDescription("a description")},
+			validate: func(t *testing.T, a *Alias) {
+				t.Helper()
+				assert.Equal(t, "valid-with-description", a.Name)
+				assert.Equal(t, "a description", a.Description)
+			},
+		},
+		{
+			name:        "host with no destination",
+			scope:       "global",
+			value:       "host.with.no.destination",
+			opts:        []Option{WithHostId("hst_1234567890")},
+			errContains: `destination_id_set_when_host_id_is_set constraint failed`,
+		},
+		{
+			name:        "unsupported project scope",
+			scope:       proj.GetPublicId(),
+			value:       "unsupported.project.scope",
+			errContains: `violates foreign key constraint "iam_scope_global_fkey"`,
+		},
+		{
+			name:        "unsupported org scope",
+			scope:       proj.GetParentId(),
+			value:       "unsupported.org.scope",
+			errContains: `violates foreign key constraint "iam_scope_global_fkey"`,
```

Comment:
This is the failure seen instead:
`        	Error:      	"db.Create: alias_must_be_in_global_scope constraint failed: check constraint violated: integrity violation: error #1000" does not contain "violates foreign key constraint \"iam_scope_global_fkey\""`